utils/ccdatofhir/ccda_to_fhir.bal

```diff
@@ -0,0 +1,186 @@
+// specific language governing permissions and limitations
+// under the License.
+
+// --------------------------------------------------------------------------------------------#
+// Source C-CDA to FHIR - Resource Mappings
+// --------------------------------------------------------------------------------------------#
+
+import ballerinax/health.fhir.r4;
+
+# Transform an C-CDA message to FHIR.
+#
+# + ccda - C-CDA message as a xml
+# + return - FHIR Bundle
+public function ccdaToFhir(xml<xml:Element|xml:Comment|xml:ProcessingInstruction|xml:Text> ccda) returns r4:Bundle|error {
+    r4:Bundle transformToFHIRResult = check transformToFhir(ccda);
+    return transformToFHIRResult;
+}
+
+function transformToFhir(xml<xml:Element|xml:Comment|xml:ProcessingInstruction|xml:Text> xmlDocument) returns r4:Bundle|error {
+
+    xmlns "urn:hl7-org:v3" as v3;
+    r4:Bundle bundle = {'type: "searchset"};
+    r4:BundleEntry[] entries = [];
+    bundle.entry = entries;
+
+    if xmlDocument is xml {
+        foreach var xmlPayload in xmlDocument {
+            if xmlPayload is xml<xml:Element> {
+                boolean isNamespaceAvailable = true;
+                xml<xml:Element|xml:Comment|xml:ProcessingInstruction|xml:Text> childElements = xmlPayload/<v3:recordTarget>;
+                if childElements.length() == 0 {
+                    isNamespaceAvailable = false;
+                    childElements = xmlPayload/<recordTarget>;
+                }
+                if childElements.length() > 0 {
+                    xml:Element|xml:Comment|xml:ProcessingInstruction|xml:Text xmlContent = childElements.get(0);
+                    if xmlContent is xml:Element {
+                        xml<xml:Element> patientRoleElement;
+                        if isNamespaceAvailable {
+                            patientRoleElement = xmlContent.children().<v3:patientRole>;
+                        } else {
+                            patientRoleElement = xmlContent.children().<patientRole>;
+                        }
+                        r4:Patient|error? mapCCDAPatientToFHIRResult = mapCcdaPatientToFhir(patientRoleElement, isNamespaceAvailable);
+                        if mapCCDAPatientToFHIRResult is error {
+                            return logAndThrowError(mapCCDAPatientToFHIRResult);
+                        }
+                        entries.push({'resource: mapCCDAPatientToFHIRResult});
+                    }
+                }
+                xml<xml:Element|xml:Comment|xml:ProcessingInstruction|xml:Text> authorElement = xmlPayload/<v3:author>;
+                if authorElement is xml:Element {
+                    r4:Practitioner|error? mapCCDAToFHIRResult = mapCcdaPractitionerToFhir(authorElement, isNamespaceAvailable);
+                    if mapCCDAToFHIRResult is error {
+                        return logAndThrowError(mapCCDAToFHIRResult);
+                    }
+                    if mapCCDAToFHIRResult.toJson() != () {
```

Comment:
`toJson()` doesn't change the value to (), why are you using `toJson()`?

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/817dadb5178bde7d809333a26f465c6688d3879a